test/spec/modules/smartxspBidAdapter_spec.js

```diff
@@ -0,0 +1,136 @@
+import {expect} from 'chai';
+import {spec} from 'modules/smartxspBidAdapter.js';
+import {newBidder} from 'src/adapters/bidderFactory.js';
+
+describe('smartxspBidAdapter', function () {
+  const adapter = newBidder(spec);
+  let bid = {
+    adUnitCode: 'adunit-code',
+    auctionId: '5kaj89l8-3456-2s56-c455-4g6h78jsdfgf',
+    bidRequestsCount: 1,
+    bidder: 'smartxsp',
+    bidderRequestId: '24081afs940568', 
+    bidderRequestsCount: 1,
+    bidderWinsCount: 0,
+    bidId: '22499d052045',
+    mediaTypes: {banner: {sizes: [[300, 250]]}},
+    params: {
+      accountId: '29Md1Mx1x2MpM7Me',
+      widgetId: 'sc.demo.oursblanc.io'
+    },
+    sizes: [
+      [300, 250]
+    ],
+    transactionId: '34562345-4dg7-46g7-4sg6-45gdsdj8fd56'
+  }
+  let bidderRequests = {
+    auctionId: 'b06c5141-fe8f-4cdf-9d7d-54415490a917',
+    auctionStart: 1579746300522,
+    bidderCode: 'myBidderCode',
+    bidderRequestId: '15246a574e859f',
+    bids: [bid],
+    refererInfo: {
+      canonicalUrl: '',
+      numIframes: 0,
+      reachedTop: true
+    }
+  }
+  describe('isBidRequestValid', function () {
+    it('should return true where required params found', function () {
+      expect(spec.isBidRequestValid(bid)).to.equal(true);
+    });
+  });
+  describe('buildRequests', function () {
+    let bidRequests = [ bid ];
+    let request = spec.buildRequests(bidRequests, bidderRequests);
+    it('sends bid request via POST', function () {
+      expect(request.method).to.equal('POST');
+    });
+    it('must contain accountId', function() {
+      expect(request.data.bidParams[0].accountId).to.equal('29Md1Mx1x2MpM7Me');
+    });
+
+    it('must contain auctionId', function() {
+      expect(request.data.auctionId).to.exist.and.to.be.a('string')
```

Comment:
this will often be null in prebid 8

Reply:
you marked this as resolved but the title of your test suggest you cannot take nulls here. If you are indeed able to, perhaps rename the the test as 'auctionid passed when available'